src/libasr/codegen/c_utils.h

```diff
@@ -527,6 +528,39 @@ class CCPPDSUtils {
             return result;
         }
 
+        std::string get_deepcopy_symbolic(ASR::expr_t *value_expr, std::string value, std::string target) {
+            std::string result;
+            if (ASR::is_a<ASR::Var_t>(*value_expr)) {
+                result = "basic_assign(" + target + ", " + value + ");";
+            } else if (ASR::is_a<ASR::IntrinsicFunction_t>(*value_expr)) {
+                ASR::IntrinsicFunction_t* intrinsic_func = ASR::down_cast<ASR::IntrinsicFunction_t>(value_expr);
+                int64_t intrinsic_id = intrinsic_func->m_intrinsic_id;
+                switch (static_cast<LCompilers::ASRUtils::IntrinsicFunctions>(intrinsic_id)) {
+                    case LCompilers::ASRUtils::IntrinsicFunctions::SymbolicSymbol: {
+                        result = "symbol_set(" + target + ", " + value + ");";
+                        break;
+                    }
+                    case LCompilers::ASRUtils::IntrinsicFunctions::SymbolicAdd: {
+                        size_t delimiterPos = value.find(",");
+                        std::string leftPart = value.substr(0, delimiterPos);
+                        std::string rightPart = value.substr(delimiterPos + 1);
+                        result = "basic_add(" + target + ", " + leftPart + ", " + rightPart + ");";
+                        break;
+                    }
+                    case LCompilers::ASRUtils::IntrinsicFunctions::SymbolicPi: {
+                        result = "basic_const_pi(" + target + ");";
+                        break;
+                    }
+                    default: {
+                        throw LCompilersException("IntrinsicFunction: `"
+                            + LCompilers::ASRUtils::get_intrinsic_name(intrinsic_id)
+                            + "` is not implemented");
+                    }
+                }
+            }
+            return result;
+        }
```

Comment:
Why not integrating this directly with `deep_copy` above?

Reply:
Well the function definitons of both `get_deepcopy` and `get_deepcopy_symbolic` are different . 
```
std::string get_deepcopy(ASR::ttype_t *t, std::string value, std::string target)
std::string get_deepcopy_symbolic(ASR::expr_t *value_expr, std::string value, std::string target)
```
We call `get_deepcopy_symbolic` once we've confirmed the `ttype`
```
if (m_target_type->type == ASR::ttypeType::SymbolicExpression){
    ASR::expr_t* m_value_expr = x.m_value;
    src += alloc + indent + c_ds_api->get_deepcopy_symbolic(m_value_expr, value, target) + "\n";
}
```
But in this function we would have to pass the `m_value` too as we are essentially checking `m_intrinsic_id` and then mapping the corresponding symengine function with it .For eg
```
SymbolicSymbol : symbol_set
SymbolicAdd : basic_add
SymbolicPi : basic_const_pi
```
We can merge them both but we might also have to pass the `m_value` in `get_deepcopy`.